tests/common/plugins/sanity_check/__init__.py

```diff
@@ -290,27 +344,26 @@ def sanity_check_full(localhost, duthosts, request, fanouthosts, nbrhosts, tbinf
 
     if not post_check:
         logger.info("No post-test check is required. Done post-test sanity check")
-        return
```

Comment:
Change to `if...else...` here because having `return` here will skip the teardown stage of `prepare_parallel_run`